pkgs/build-support/trivial-builders/default.nix

```diff
@@ -347,12 +347,20 @@ rec {
       '';
 
       checkPhase =
+        # GHC (=> shellcheck) isn't supported on some platforms (such as risc-v)
+        # but we still want to use writeShellApplication on those platforms
+        let
+          shellcheckSupported = (builtins.tryEval ghc.bootPkgs.ghc).success;
```

Comment:
Don't you mean `buildPlatform`? Also this evaluates to `true`. I think it's because the platform support in the ghc expression is implemented with `throw`

Reply:
Can we convince the ghc people to change that to meta.platforms? Maybe it is super simple because they generate many things.

Reply:
@NixOS/haskell

Reply:
All the checks proposed here are (slightly) incorrect. That using `meta.platforms` is not possible is a slight oversight, namely that (shallowly) evaluating `passthru` (which `mkDerivation` needs to do) forces `binDistUsed`. I'll correct that separately and then it should be possible again to use `meta.platforms`.

The correct check for `shellcheck`'s availability is `lib.meta.availableOn stdenv.buildPlatform shellcheck.compiler` which checks if the compiler we *would* use to build `shellcheck` itself is available. I need to warn you though that `meta.platforms` is imperfect, so there may be some edge cases, but I'm not sure if this applies to any platforms `stdenv` supports as a build platform. This is improved in #212188.